### PR TITLE
Create initial structure and gcloud implementation

### PR DESCRIPTION
**Pull request by boekkooi-fresh**
This PR implements [OpenTracing](http://opentracing.io/) into PHP and have a BasicTracer and Google cloud implementation.

### CODE REVIEW

File: src/BasicTracer/Propagation/TextMapPropagator.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    $traceId = $value;
+                    break;
+                case self::FIELD_SPAN_ID:
+                    $spanId = (int) $value;
+                    break;
+                case self::FIELD_SAMPLED:
+                    $sampled = (bool) $value;
+
+                    break;
+                case substr($keyLower, 0, strlen(self::PREFIX_BAGGAGE)) === self::PREFIX_BAGGAGE:
```
Comment (nenadstojanovikj) [inline src/BasicTracer/Propagation/TextMapPropagator.php]: PHPStorm suggests that could use `strpos($keyLower, self::PREFIX_BAGGAGE) === 0` to improve readability, which I agree.

**Comment by nenadstojanovikj** [comment]
:+1:

File: src/BasicTracer/Span.php [hunk 2]
```diff
@@ -196,7 +196,9 @@ public function getBaggageItem(string $key)
     private function assertUnfinished()
     {
         if ($this->recorder === null) {
-            throw new \LogicException('You are the destroyer of worlds! Maybe it\'s time to go to bed?');
+            throw new SpanStateException(
+                'The Span is finished! You have become the destroyer of worlds! Maybe it\'s time to go to bed?'
```
Comment (nenadstojanovikj) [inline src/BasicTracer/Span.php]: :sleeping_bed: :zzz:

**Comment by nenadstojanovikj** [comment]
:+1:

**Comment by emilyclaudet** [comment]
👍

**Pull request merged**